torch/csrc/PyInterpreter.cpp

```diff
@@ -40,6 +40,16 @@ namespace {
     }                                                                         \
   }
 
+py::object create_self_python_object(const c10::TensorImpl* self) {
+  auto self_t =
+      TensorBase(c10::intrusive_ptr<c10::TensorImpl, c10::UndefinedTensorImpl>::
+                     // NOLINTNEXTLINE(cppcoreguidelines-pro-type-const-cast)
+                 reclaim(const_cast<c10::TensorImpl*>(self)));
+  auto self_p = py::reinterpret_steal<py::object>(THPVariable_Wrap(self_t));
+  self_t.unsafeReleaseTensorImpl();
```

Comment:
According to https://github.com/pytorch/pytorch/blob/e885225edaf94887d483ed7656150c8b5a819bc1/c10/util/intrusive_ptr.h#L463-L464 this makes self_t invalid?
I will defer to @ezyang here for the best way to go.

Reply:
Yes, as the comment says, to avoid decrease the refcount.